upload/tests/test_upload.py

```diff
@@ -2172,11 +2172,57 @@ def test_azure_no_server_uri(self):
             line.strip() for line in expected_error.split("\n")
         ]
 
+    def test_azure_invalid_server_uri(self):
+        expected_error = """Unable to locate build via Azure API. Please upload with the Codecov repository upload token to resolve issue."""
+
+        params = {
+            "project": "project123",
+            "job": 732059764,
+            "server_uri": "https://dev.azure.com/missing_trailing_slash",
+        }
+        with pytest.raises(NotFound) as e:
+            TokenlessUploadHandler("azure_pipelines", params).verify_upload()
+        assert [line.strip() for line in e.value.args[0].split("\n")] == [
+            line.strip() for line in expected_error.split("\n")
+        ]
+
+        params["server_uri"] = "https://missing_trailing_slash.visualstudio.com"
```

Comment:
this test looks like it would benefit from becoming a `pytest.mark.parametrize` test given its repeating nature